src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,129 @@
+     */
+    public static Task parseData(String s) throws DukeException {
+        try {
+            String[] strings = s.split(" \\| ");
+            Task task;
+
+            switch (strings[0]) {
+            case "T":
+                if (strings.length > 3) {
+                    throw new DukeException("Invalid task data loaded.");
+                }
+                task = new Todo(strings[2]);
+                break;
+            case "D":
+                if (strings.length > 4) {
+                    throw new DukeException("Invalid task data loaded.");
+                }
+                task = new Deadline(strings[2], strings[3]);
+                break;
+            case "E":
+                if (strings.length > 4) {
+                    throw new DukeException("Invalid task data loaded.");
+                }
+                task = new Event(strings[2], strings[3]);
+                break;
+            default:
+                throw new DukeException("Invalid task data loaded.");
+            }
+
+            if (strings[1].equals("X")) {
+                task.markAsDone();
+            } else if (!strings[1].equals(" ")) {
+                throw new DukeException("Invalid task data loaded.");
+            }
+
+            return task;
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new DukeException("Invalid task data loaded.");
+        }
+    }
+
+    /**
+     * Marks a Task as done.
+     */
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    /**
+     * Marks a Task as not done.
+     */
+    public void unmarkAsDone() {
+        isDone = false;
+    }
+
+    /**
+     * Returns an icon that represents if the task is done or not.
+     *
+     * @return "X" if the task is done, " " if not.
+     */
+    public String getStatusIcon() {
+        return isDone ? "X" : " ";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String toString() {
+        return "[" + getStatusIcon() + "] " + description;
+    }
+
+    /**
+     * Returns the representation of a task when stored in a data file on the hard disk.
+     *
+     * @return a string representing the task as it is stored on a data file on the hard disk.
+     */
+    public String toData() {
+        return getStatusIcon() + " | " + description;
+    }
+
+    /**
+     * Checks if the description of a task contains a specified keyword (or phrase).
+     *
+     * @param keyword The specified keyword (or phrase) to check.
+     * @return true if the task description contains the specified keyword (or phrase).
+     */
+    public boolean contains(String keyword) {
+        return description.contains(keyword);
+    }
+
+    /**
+     * Checks if a Task occurs by/at a specific date.
+     *
+     * @param date The specified date to check.
+     * @return true if the task occurs by/at the specified date.
+     */
+    public abstract boolean onDate(LocalDate date);
```

Comment:
Maybe naming it as isOnDate makes it clearer that this method returns a boolean?